tests/validation/tests/v3_api/test_hpa.py

```diff
@@ -0,0 +1,499 @@
+    # Verify that the Read Only member can list the HPA
+    hpa = create_hpa(cluster_owner_p_client, ns)
+    hpaname = hpa["name"]
+    hpadict = readonly_user_client.list_horizontalPodAutoscaler(name=hpaname)
+    print(hpadict)
+    assert hpa.type == "horizontalPodAutoscaler"
+    hpadata = hpadict.get('data')
+    assert len(hpadata) == 1
+    assert hpadata[0].name == hpaname
+    remove_resource(hpa)
+
+
+@if_test_rbac
+def test_rbac_hpa_project_readonly_delete(remove_resource):
+    project = rbac_get_project()
+    ns = rbac_get_namespace()
+    cluster_owner_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
+    cluster_owner_p_client = get_project_client_for_token(project,
+                                                          cluster_owner_token)
+    user_token = rbac_get_user_token_by_role(PROJECT_READ_ONLY)
+    readonly_user_client = get_project_client_for_token(project, user_token)
+
+    # As a Cluster owner create a HPA object
+    hpa = create_hpa(cluster_owner_p_client, ns)
+
+    # Verify that the Read Only member cannot delete the HPA objects created by
+    # Cluster Owner
+    with pytest.raises(ApiError) as e:
+        delete_hpa(readonly_user_client, hpa, ns)
+    assert e.value.error.status == 403
+    assert e.value.error.code == 'Forbidden'
+    remove_resource(hpa)
+
+
+def create_hpa(p_client, ns):
+
+    # Create workload of scale 1 with CPU reservation
+    # Create hpa pointing to the workload.
+
+    con = [{"name": "test1",
+            "image": TEST_IMAGE,
+            "resources": {
+                "requests": {
+                    "memory": "128Mi",
+                    "cpu": "100m"
+                },
+                "limits": {
+                    "memory": "256Mi",
+                    "cpu": "1000m"
+                }
+            }
+            }]
+    name = random_test_name("default")
+    workload = p_client.create_workload(name=name,
+                                        containers=con,
+                                        namespaceId=ns.id)
+    assert workload.scale == 1
+    validate_workload(p_client, workload, "deployment", ns.name)
+    metrics = [{
+            'name': 'cpu',
+            'type': 'Resource',
+            'target': {
+                'type': 'Utilization',
+                'utilization': '50',
+            },
+        }]
+
+    hpa = p_client.create_horizontalPodAutoscaler(
+        name=name,
+        namespaceId=ns.id,
+        maxReplicas=5,
+        minReplicas=2,
+        workloadId=workload.id,
+        metrics=metrics
+    )
+    wait_for_hpa_to_active(p_client, hpa)
+    assert hpa.type == "horizontalPodAutoscaler"
+    assert hpa.name == name
+    # After hpa becomes active, the workload scale should be equal to the
+    # minReplicas set in HPA object
+    workloadlist = p_client.list_workload(uuid=workload.uuid).data
+    assert workloadlist[0].scale == hpa.minReplicas
+
+    return hpa
```

Comment:
Should we return the workload as well? So we can remove both hpa and the workload later.